state/accumulator.go

```diff
@@ -212,6 +212,33 @@ func (a *Accumulator) Initialise(roomID string, state []json.RawMessage) (Initia
 		if len(events) == 0 {
 			return fmt.Errorf("failed to insert events, all events were filtered out: %w", err)
 		}
+
+		// Before proceeding further, ensure that we have "proper" state and not just a
+		// single stray event by looking for the create event.
+		hasCreate := false
+		for _, e := range events {
+			if e.Type == "m.room.create" && e.StateKey == "" {
+				hasCreate = true
+				break
+			}
+		}
+		if !hasCreate {
+			const errMsg = "cannot create first snapshot without a create event"
+			sentry.WithScope(func(scope *sentry.Scope) {
+				scope.SetContext(internal.SentryCtxKey, map[string]interface{}{
+					"room_id":   roomID,
+					"len_state": len(events),
+				})
+				sentry.CaptureMessage(errMsg)
+			})
+			logger.Warn().
+				Str("room_id", roomID).
+				Int("len_state", len(events)).
+				Msg(errMsg)
+			return fmt.Errorf(errMsg)
```

Comment:
This should have been a DataError. Otherwise the poller will end up retrying the same sync again and again.